client.go

```diff
@@ -37,8 +50,85 @@ func defaultSocketName() string {
 	return path.Join(os.TempDir(), fmt.Sprintf("emacs%d", os.Getuid()), "server")
 }
 
+// defaultEmacsDir returns the default Emacs configuration directory (aka `.emacs.d`) for the current user.
+func defaultEmacsDir() string {
+	var emacsDirName string // default folder name used by emacs
+	var emacsDir string
+
+	if runtime.GOOS == "windows" {
+		emacsDirName = ".emacs.d" // windows always uses '.emacs.d'
+	} else {
+		emacsDirName = "emacs"
+	}
+
+	xdgPathA := path.Join(os.Getenv("XDG_CONFIG_HOME"), emacsDirName)
+	xdgPathB := path.Join(os.Getenv("HOME"), ".config", emacsDirName) // user following convention without XDG_CONFIG_HOME set
+	legacyPath := path.Join(os.Getenv("HOME"), emacsDirName)          // windows (if HOME is set) and emacs pre-v27
+
+	userConfigDir, _ := os.UserConfigDir()
+	osDefaultPath := path.Join(userConfigDir, emacsDirName)
+
+	switch {
+	case checkPath(xdgPathA):
+		emacsDir = xdgPathA
+	case checkPath(xdgPathB):
+		emacsDir = xdgPathB
+	case checkPath(legacyPath):
+		emacsDir = legacyPath
+	case checkPath(osDefaultPath):
+		emacsDir = osDefaultPath
+	}
+
+	return emacsDir
+}
+
+// defaultServerFile returns the default Emacs server file for the current user.
+func defaultServerFile() (serverFile string) {
+	fromEnv := os.Getenv("EMACS_SERVER_FILE")
+	fromEmacsDir := path.Join(defaultEmacsDir(), "server", "server")
+
+	if fromEnv != "" {
+		serverFile = fromEnv
+	} else {
+		serverFile = fromEmacsDir
+	}
+
+	return
+}
+
+// parseServerFile return the emacs server TCP address and auth key from an emacs server file
+func parseServerFile(serverFile string) (serverAddr string, authString string) {
+
+	fp, err := os.Open(serverFile)
+	if err != nil {
+		panic(err)
+	}
+	defer fp.Close()
+
+	scanner := bufio.NewScanner(fp)
+	if scanner.Scan() {
+		// 1st line
+		serverAddr = strings.Split(scanner.Text(), " ")[0]
+	}
+	if scanner.Scan() {
+		// 2nd line
+		authString = scanner.Text()
+	}
+
+	return
+}
+
 // Dial connects to the remote Emacs server.
 func Dial(options *Options) (net.Conn, error) {
+	if checkPath(options.SocketName) {
+		return dialUnix(options)
+	} else {
```

Comment:
It would be a good idea to check here that the serverPath exists and if it doesn't return an error that says that no emacs server is available.